zephyr/src/utils.c

```diff
@@ -656,7 +661,22 @@ int send_broadcast_arp(char *target_ip, int *send_count, int rate) {
 }
 
 int find_interface_ip(char *ipaddr, int ipaddr_len, char *name) {
-    return 0;
+    struct wpa_supplicant *wpa_s;
+    char tmp[32];
+
+    if (strcmp(name, get_wireless_interface()))
+        return 0;
+    wpa_s = z_wpas_get_handle_by_ifname(name);
+    if (!wpa_s) {
+        indigo_logger(LOG_LEVEL_ERROR, "Unable to find the interface: %s, quitting", name);
+        return 0;
+    }
+
+    if (wpa_s->l2 && l2_packet_get_ip_addr(wpa_s->l2, tmp, sizeof(tmp)) >= 0) {
```

Comment:
Please note this only gives you IPV4 address

Reply:
@krish2718   Made the changes to read both ipv4 and ipv6 addresses, kindly review.